src/Sentry/SentryFailedRequestHandler.cs

```diff
@@ -0,0 +1,95 @@
+using System.Net;
+using System.Reflection.PortableExecutable;
+using Sentry.Internal;
+
+namespace Sentry;
+
+internal class SentryFailedRequestHandler : ISentryFailedRequestHandler
+{
+    private readonly IHub _hub;
+    private readonly SentryOptions _options;
+
+    public const string ResponseKey = "response";
+    public const string MechanismType = "SentryFailedRequestHandler";
+
+    internal SentryFailedRequestHandler(IHub hub, SentryOptions options)
+    {
+        _hub = hub;
+        _options = options;
+    }
+
+    public void HandleResponse(HttpResponseMessage response)
+    {
+        // Ensure reponse and request are not null
+        if (response?.RequestMessage is null)
+        {
+            return;
+        }
+
+        // Don't capture if the option is disabled
+        if (_options?.CaptureFailedRequests is false)
+        {
+            return;
+        }
+
+        // Don't capture events for successful requets
+        if (_options?.FailedRequestStatusCodes.Any(range => range.Contains(response.StatusCode)) is false)
+        {
+            return;
+        }
+
+        // Ignore requests to the Sentry DSN
+        var uri = response.RequestMessage.RequestUri;
+        if (_options?.Dsn is { } dsn && new Uri(dsn).Host.Equals(uri?.Host, StringComparison.OrdinalIgnoreCase))
+        {
+            return;
+        }
+
+        // Ignore requests that don't match the FailedRequestTargets
+        var requestString = uri?.OriginalString ?? "";
+        if (_options?.FailedRequestTargets.ContainsMatch(requestString) is false)
+        {
+            return;
+        }
+
+        // Capture the event
+        try
+        {
+            response.EnsureSuccessStatusCode();
+        }
+        catch (HttpRequestException exception)
+        {
+            exception.SetSentryMechanism(MechanismType);
+
+            var @event = new SentryEvent(exception);
+
+            var sentryRequest = new Request
```

Comment:
@mattjohnsonpint excepting hints, this implementation now mimics [what the Java SDK is doing](https://github.com/getsentry/sentry-java/blob/0be0d10a9fc61369a551c3b90134fe36c050a584/sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt#L183). The Java SDK sets the `event.Request` and `event.Contexts["response"]` properties. However it only includes cookies and headers if `SendDefaultPii == true`.

Reply:
Sounds good.  Thanks!